### PR TITLE
Upgrade Jquery to version 3.6

### PR DESCRIPTION
**Pull request by AxelBolle**
This will make ninja work with the latest version of Jquery.

Our quicklink implementation had to be rewritten to exclude a deprecated ui-library.

A lot of repositories depend on ninjas JavaScript and may well fail to build if this is merged. 
This merge is however needed to make other repos able to use ninjas jQuery 3.6.

This might be best reviewed in a screen sharing session.

### CODE REVIEW

File: application/src/js/application/common.js [hunk 1]
```diff
@@ -243,80 +243,151 @@ $(document).ready(function() {
...
-				$('#dojo-add-quicklink-icon').attr('value','');
-			}
-		});
-	}
-
-	$(window).on('load', function () {
-		$('#dojo-icon-container').on('click', 'span', function() {
-			var span = $(this);
-			$('#dojo-add-quicklink-icon').val(span.data('icon'));
+		/* QUICKLINK EXTENSION */
```
[the remark this answers is not in the document]
Comment (AxelBolle) [reply application/src/js/application/common.js]: Fixed

File: application/src/js/application/common.js [hunk 2]
```diff
@@ -243,80 +243,151 @@ $(document).ready(function() {
...
+
+		// Saves the quicklink to the user settings
+		function quicklink_save(lightbox) {
+			var href = $('#dojo-add-quicklink-href').prop('value'),
+				title = $('#dojo-add-quicklink-title').prop('value'),
+				icon = $('#dojo-add-quicklink-icon').prop('value'),
+				target = $('#dojo-add-quicklink-target').prop('value'),
 				changed = false;
-			var error = '';
+			let error = '';
```
[the remark this answers is not in the document]
Comment (AxelBolle) [reply application/src/js/application/common.js]: Fixed

File: application/src/js/jquery.contextMenu.js [hunk 3]
```diff
@@ -185,9 +185,12 @@ if(jQuery)( function() {
 		});
 
 		// Disable text selection
-		if( $.browser.mozilla ) {
+		navigatorObject = window.navigator
+		var UsrAg = navigator.userAgent;
+
+		if(UsrAg.indexOf("Firefox") > -1) {
 			$('#' + o.menu).each( function() { $(this).css({ 'MozUserSelect' : 'none' }); });
```
[the remark this answers is not in the document]
Comment (AxelBolle) [reply application/src/js/jquery.contextMenu.js]: Removed

File: application/src/js/jquery.timePicker.js [hunk 4]
```diff
@@ -223,10 +223,6 @@
 		elm.value = $(sel).text();
 		// Trigger element's change events.
 		$(elm).change();
-		// Keep focus for all but IE (which doesn't like it)
-		if (!$.browser.msie) {
-			elm.focus();
```
[the remark this answers is not in the document]
Comment (AxelBolle) [reply application/src/js/jquery.timePicker.js:228]: Yes, fixed.

File: modules/lsfilter/src/js/LSFilterList.js [hunk 5]
```diff
@@ -27,8 +27,10 @@ function lsfilter_list(config)
 		this.config.autorefresh_enabled = false;
 	}
 
-	if($.browser.msie) {
-		var parts = $.browser.version.split('.');
+	if (window.document.documentMode) {
+		var ua = window.navigator.userAgent;
+		var msie = ua.indexOf('MSIE ');
+		var parts = msie.split('.');
```
[the remark this answers is not in the document]
Comment (AxelBolle) [reply modules/lsfilter/src/js/LSFilterList.js]: Removed

File: modules/lsfilter/src/js/LSFilterInputWindow.js [hunk 6]
```diff
@@ -19,7 +19,10 @@ $().ready(function() {
 			return false;
 		}
 	});
-	$(document).on("hover", "#filter_result table td a span.x16-add-comment", function() {
+	$(document).on("mouseenter", "#filter_result table td a span.x16-add-comment", function() {
 		$(this).removeAttr("title");
 	});
+	$(document).on("mouseleave", "#filter_result table td a span.x16-add-comment", function() {
+		$(this).prop("title");
```
[the remark this answers is not in the document]
Comment (AxelBolle) [reply modules/lsfilter/src/js/LSFilterInputWindow.js]: Removed, keeping previous behaviour.

File: modules/lsfilter/src/js/LSFilterInputWindow.js [hunk 7]
```diff
@@ -19,7 +19,7 @@ $().ready(function() {
 			return false;
 		}
 	});
-	$(document).on("hover", "#filter_result table td a span.x16-add-comment", function() {
+	$(document).on("mouseenter", "#filter_result table td a span.x16-add-comment", function() {
```
Comment (jimorie) [inline modules/lsfilter/src/js/LSFilterInputWindow.js:22]: I don't understand the purpose of this event listener. But, I mean, sure, considering what it does I guess this is equivalent.

**Review by jimorie — APPROVED** [review]
Nice! Looks good to me. :+1:

**Pull request merged**